backend/uclapi/resources/views.py

```diff
@@ -0,0 +1,52 @@
+import os
+import requests
+
+from lxml import etree
+
+from common.decorators import uclapi_protected_endpoint
+from common.helpers import PrettyJsonResponse as JsonResponse
+
+from rest_framework.decorators import api_view
+
+
+@api_view(['GET'])
+@uclapi_protected_endpoint()
+def get_pc_availability(request, *args, **kwargs):
+    try:
+        r = requests.get(os.environ["PCA_LINK"])
+    except requests.exceptions.MissingSchema:
+        return JsonResponse({
+            "ok": False,
+            "error": "Couldn't get the availability data. Try again later"
+        }, rate_limiting_data=kwargs)
+
+    try:
+        e = etree.fromstring(r.content)
+    except (ValueError, etree.XMLSyntaxError):
+        return JsonResponse({
+            "ok": False,
+            "error": "Couldn't parse the availability data"
```

Comment:
`Could not parse the desktop availability data. Please try again later or contact us for support.`